src/github.com/getlantern/lantern-mobile/app/src/main/java/org/getlantern/lantern/activity/LanternMainActivity.java

```diff
@@ -104,57 +182,289 @@ protected void onCreate(Bundle savedInstanceState) {
+                        R.drawable.ic_contact));
+        }};
+
+        final ListAdapter listAdapter = new ListAdapter(this, mNavItems);  
+
+        if (session.showFeed())  {
+            // 'Turn off Feed' when the feed is already shown
+            mNavItems.add(new NavItem(resources.getString(R.string.newsfeed_off_option), R.drawable.ic_feed));
+        } else {
+            // 'Try Lantern Feed' when the feed is already hidden
+            mNavItems.add(new NavItem(resources.getString(R.string.newsfeed_option), R.drawable.ic_feed));
+        }
+        
+        mNavItems.add(new NavItem(resources.getString(R.string.quit_option), 
+                    R.drawable.ic_quit));
 
-        } catch (Exception e) {
-            Log.d(TAG, "Got an exception " + e);
+        menuMap.put(resources.getString(R.string.quit_option), new Command() { 
+            public void runCommand() { quitLantern(); } 
+        });
+
+        menuMap.put(resources.getString(R.string.contact_option), new Command() { 
+            public void runCommand() { contactOption(); } 
+        });
+
+        menuMap.put(resources.getString(R.string.newsfeed_off_option), new Command() {
+            public void runCommand() {
+                updateFeedview(listAdapter, mNavItems, resources, 3, false);
+            }
+        });
+
+        menuMap.put(resources.getString(R.string.newsfeed_option), new Command() {
+            public void runCommand() {
+                updateFeedview(listAdapter, mNavItems, resources, 3, true);
+            }
+        });
+
+        menuMap.put(resources.getString(R.string.desktop_option), new Command() { 
+            public void runCommand() {
+                startActivity(new Intent(activity, DesktopActivity_.class));
+            } 
+        });
+
+        menuMap.put(resources.getString(R.string.share_option), new Command() { 
+            public void runCommand() {
+                final Shareable shareable = new Shareable(activity);
+                if (shareable != null) {
+                    shareable.showOption();
+                }
+            }
+        });   
+
+        // Populate the Navigtion Drawer with options
+        drawerList.setAdapter(listAdapter);
+
+        // remove ListView border
+        drawerList.setDivider(null);
+
+        // Drawer Item click listeners
+        drawerList.setOnItemClickListener(new AdapterView.OnItemClickListener() {
+            @Override
+            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
+
+                drawerList.setItemChecked(position, true);
+                if (position >= 0 && position < mNavItems.size()) {
+                    NavItem item = mNavItems.get(position);
+                    if (item != null) {
+                        String title = item.getTitle();
+                        Log.d(TAG, "Menu option " + title + " selected");
+                        menuMap.get(title).runCommand();
```

Comment:
Do we need to make sure `menuMap.get(title)` doesn't return null here? And maybe log an error if it ever does as well as if `mNavItems.get(position)` ever returns `null` above that?